service/src/test/java/uk/nhs/adaptors/gp2gp/uat/EhrExtractUATTest.java

```diff
@@ -0,0 +1,165 @@
+    private static final String INPUT_JSON_WITH_IMMUNISATIONS = INPUT_JSON_WITH_STRUCTURED_CONSULTATIONS_INCLUDING_NARRATIVE_CONTENT;
+    private static final String INPUT_JSON_WITH_DRUG_AND_NON_DRUG_ALLERGIES = "9465700827_Bentley_allergy1only_20210212.json";
+    private static final String INPUT_JSON_WITH_DIARY_ENTRIES = "9465700827_Bentley_diary1only_20210212.json";
+    private static final String INPUT_JSON_WITH_REFERRALS = "9465701149_Drake_refs1only_20210212.json";
+    private static final String INPUT_JSON_WITH_PROBLEMS_AND_PROBLEM_LINKAGES = "9465699918_Magre_prob1only_20210212.json";
+    private static final String INPUT_JSON_WITH_NON_SNOMED_CODED_DATA_VIA_EGTON_CODES = "9465701483_Dougill_uncat18jan_20210215.json";
+    private static final String OUTPUT_XML_USES_PATHOLOGY_RECORD = "9465701718_Guerra_inv1only_20210212.xml";
+    private static final String OUTPUT_XML_USES_MEDICATION_RECORD = "9465701645_Prytherch_medsonly_20210212.xml";
+    private static final String OUTPUT_XML_USES_STRUCTURED_CONSULTATIONS_INCLUDING_NARRATIVE_CONTENT =
+        "9465701610_Powne_imms1only_20210212.xml";
+    private static final String OUTPUT_XML_USES_GENERAL_MISCELLANEOUS_CODED_OBSERVATION_CONTENT = OUTPUT_XML_USES_PATHOLOGY_RECORD;
+    private static final String OUTPUT_XML_USES_BLOOD_PRESSURES = "9465701610_Powne_cons1only_20210212.xml";
+    private static final String OUTPUT_XML_USES_IMMUNISATIONS = OUTPUT_XML_USES_STRUCTURED_CONSULTATIONS_INCLUDING_NARRATIVE_CONTENT;
+    private static final String OUTPUT_XML_USES_DRUG_AND_NON_DRUG_ALLERGIES = "9465700827_Bentley_allergy1only_20210212.xml";
+    private static final String OUTPUT_XML_USES_DIARY_ENTRIES = "9465700827_Bentley_diary1only_20210212.xml";
+    private static final String OUTPUT_XML_USES_REFERRALS = "9465701149_Drake_refs1only_20210212.xml";
+    private static final String OUTPUT_XML_USES_PROBLEMS_AND_PROBLEM_LINKAGES = "9465699918_Magre_prob1only_20210212.xml";
+    private static final String OUTPUT_XML_USES_NON_SNOMED_CODED_DATA_VIA_EGTON_CODES = "9465701483_Dougill_uncat18jan_20210215.xml";
+    private static final String TEST_ID_1 = "test-id-1";
+    private static final String TEST_ID_2 = "test-id-2";
+    private static final String TEST_ID_3 = "test-id-3";
+    private static final String TEST_CONVERSATION_ID = "test-conversation-id";
+    private static final String TEST_REQUEST_ID = "test-request-id";
+    private static final String TEST_NHS_NUMBER = "1234567890";
+    private static final String TEST_FROM_ODS_CODE = "test-from-ods-code";
+    private static final String TEST_TO_ODS_CODE = "test-to-ods-code";
+    private static final String TEST_DATE_TIME = "2020-01-01T01:01:01.01Z";
+
+    private static GetGpcStructuredTaskDefinition getGpcStructuredTaskDefinition;
+
+    @Mock
+    private RandomIdGeneratorService randomIdGeneratorService;
+    @Mock
+    private TimestampService timestampService;
+    @Mock
```

Comment:
I don't think we need the Mock annotation for codeableConceptCdMapper

Reply:
If I remove the @mock above the codeableConceptCdMapper declaration, one test will not pass due to the NullPointerException, so seems that it might be needed. What do you think?

Reply:
As has been suggested by you during our call. Here is a change: fc659f6ece666e8a34e241e4b2139509acbf6c6e
Thanks!